src/app/sign-up-get-starter-deso/sign-up-get-starter-deso.component.ts

```diff
@@ -0,0 +1,259 @@
+  submittingPhoneNumberVerificationCode = false;
+  screenToShow: string | null = null;
+  SignUpGetStarterDESOComponent = SignUpGetStarterDESOComponent;
+  phoneNumber: string = "";
+  phoneNumberCountryCode: string | null = null;
+  resentVerificationCode = false;
+  sendPhoneNumberVerificationTextServerErrors = new SendPhoneNumberVerificationTextServerErrors();
+  submitPhoneNumberVerificationCodeServerErrors = new SubmitPhoneNumberVerificationCodeServerErrors();
+  user: User | null = null;
+  loading: boolean = true;
+  isPhoneNumberSuccess: boolean = false;
+
+  constructor(
+    public globalVars: GlobalVarsService,
+    private backendApi: BackendAPIService,
+    private activatedRoute: ActivatedRoute,
+    private identityService: IdentityService,
+    private accountService: AccountService,
+  ) {}
+
+  ngOnInit(): void {
+    this._setScreenToShow();
+    this.activatedRoute.queryParams.subscribe(params => {
+      if (params.public_key) {
+        this.publicKey = params.public_key;
+        this.backendApi.GetUsersStateless([this.publicKey]).subscribe((res) => {
+          if (res.UserList?.length) {
+            this.user = res.UserList[0];
+          }
+        }, (err) => {
+          console.error(err);
+        }).add(() => this.loading = false);
+      } else {
+        this.loading = false;
+      }
+    });
+
+  }
+
+  _setScreenToShow() {
+    // TODO: refactor silly setInterval
+    let interval = setInterval(() => {
+      if (this.user?.HasPhoneNumber == null) {
+        // Wait until we've loaded the HasPhoneNumber boolean from the server
+        return;
+      }
+
+      if (this.user?.HasPhoneNumber) {
+        this.screenToShow = SignUpGetStarterDESOComponent.COMPLETED_PHONE_NUMBER_VERIFICATION_SCREEN;
+      } else {
+        this.screenToShow = SignUpGetStarterDESOComponent.CREATE_PHONE_NUMBER_VERIFICATION_SCREEN;
+      }
+
+      clearInterval(interval);
+    }, 50);
+  }
+
+  backToPreviousSignupStepOnClick() {
+    this.backToPreviousSignupStepClicked.emit();
+  }
+
+  backButtonClickedOnSubmitVerificationScreen() {
+    this.screenToShow = SignUpGetStarterDESOComponent.CREATE_PHONE_NUMBER_VERIFICATION_SCREEN;
+  }
+
+  sendVerificationText() {
+    if (this.phoneForm.invalid) {
+      return;
+    }
+
+    this._sendPhoneNumberVerificationText();
+  }
+
+  resendVerificationCode(event: Event) {
+    event.stopPropagation();
+    event.preventDefault();
+
+    // Return if the user just resent the verification code (to prevent multiple unnecessary texts)
+    if (this.resentVerificationCode) {
+      return false;
+    }
+
+    // Clear any existing resend-related errors
+    this.sendPhoneNumberVerificationTextServerErrors = new SendPhoneNumberVerificationTextServerErrors();
+
+    this._sendPhoneNumberVerificationText();
+
+    // Display a success indicator
+    this.resentVerificationCode = true;
+    setTimeout(() => (this.resentVerificationCode = false), 5000);
```

Comment:
This is working, right? Was just thinking if we might need a `.bind(this)` here, but it's probably fine... `this` scopes always confuse me.

Reply:
This thing `()=>{}` keeps `this` as whatever it was in the parent scope. It's generally always what you want.